src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1671,6 +1618,24 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
             // UNDONE: Otherwise if one member is a non-lifted operator and  the other is a lifted
             // operator, the non-lifted one is better.
 
+            // Not spec'd yet: if two members are static extension methods, the more specific type that they
+            // are extending wins (even though it appears nowhere in the signature).
+            if (m1.Member.IsInExtensionClass && m2.Member.IsInExtensionClass && !m1.Member.IsUnreducedExtensionMember && !m2.Member.IsUnreducedExtensionMember)
+            {
+                var m1type = m1.Member.ContainingType.ExtensionClassType;
+                var m2type = m2.Member.ContainingType.ExtensionClassType;
+                if (Conversions.HasIdentityOrImplicitReferenceConversion(m1type, m2type, ref useSiteDiagnostics) ||
+                    Conversions.HasIdentityOrImplicitReferenceConversion(m2type, m1type, ref useSiteDiagnostics))
```

Comment:
Do we choose more specific types for value types as well? `extension class A : object` and `extension class B : int`